sdk/core/azure-core/test/ut/http_method.cpp

```diff
@@ -0,0 +1,288 @@
+
+  EXPECT_EQ(delete_, HttpMethod::Delete);
+  EXPECT_EQ(HttpMethod::Delete, delete_);
+
+  EXPECT_NE(delete_, HttpMethod::Get);
+  EXPECT_NE(delete_, HttpMethod::Head);
+  EXPECT_NE(delete_, HttpMethod::Post);
+  EXPECT_NE(delete_, HttpMethod::Put);
+  EXPECT_NE(delete_, HttpMethod::Patch);
+  EXPECT_NE(delete_, HttpMethod("TRACE"));
+  EXPECT_NE(delete_, HttpMethod("HUMUHUMUNUKUNUKUAPUAA"));
+
+  EXPECT_NE(HttpMethod::Get, delete_);
+  EXPECT_NE(HttpMethod::Head, delete_);
+  EXPECT_NE(HttpMethod::Post, delete_);
+  EXPECT_NE(HttpMethod::Put, delete_);
+  EXPECT_NE(HttpMethod::Patch, delete_);
+  EXPECT_NE(HttpMethod("TRACE"), delete_);
+  EXPECT_NE(HttpMethod("HUMUHUMUNUKUNUKUAPUAA"), delete_);
+}
+
+TEST(HttpMethod, Patch)
+{
+  HttpMethod const patch = HttpMethod::Patch;
+  EXPECT_EQ(patch.ToString(), "PATCH");
+
+  EXPECT_EQ(HttpMethod::Patch, HttpMethod::Patch);
+  EXPECT_EQ(patch, patch);
+  EXPECT_EQ(patch, HttpMethod("PATCH"));
+  EXPECT_EQ(HttpMethod("PATCH"), HttpMethod::Patch);
+
+  EXPECT_NE(patch, HttpMethod("Patch"));
+  EXPECT_NE(HttpMethod("Patch"), patch);
+
+  EXPECT_NE(patch, HttpMethod(std::string()));
+  EXPECT_NE(HttpMethod(std::string()), patch);
+
+  EXPECT_EQ(patch, HttpMethod::Patch);
+  EXPECT_EQ(HttpMethod::Patch, patch);
+
+  EXPECT_NE(patch, HttpMethod::Get);
+  EXPECT_NE(patch, HttpMethod::Head);
+  EXPECT_NE(patch, HttpMethod::Post);
+  EXPECT_NE(patch, HttpMethod::Put);
+  EXPECT_NE(patch, HttpMethod::Delete);
+  EXPECT_NE(patch, HttpMethod("TRACE"));
+  EXPECT_NE(patch, HttpMethod("HUMUHUMUNUKUNUKUAPUAA"));
+
+  EXPECT_NE(HttpMethod::Get, patch);
+  EXPECT_NE(HttpMethod::Head, patch);
+  EXPECT_NE(HttpMethod::Post, patch);
+  EXPECT_NE(HttpMethod::Put, patch);
+  EXPECT_NE(HttpMethod::Delete, patch);
+  EXPECT_NE(HttpMethod("TRACE"), patch);
+  EXPECT_NE(HttpMethod("HUMUHUMUNUKUNUKUAPUAA"), patch);
+}
+
+TEST(HttpMethod, Trace)
+{
+  HttpMethod const trace = HttpMethod("TRACE");
+  EXPECT_EQ(trace.ToString(), "TRACE");
+
+  EXPECT_EQ(trace, trace);
+  EXPECT_EQ(trace, HttpMethod("TRACE"));
+  EXPECT_EQ(HttpMethod("TRACE"), trace);
+
+  EXPECT_NE(trace, HttpMethod("Trace"));
+  EXPECT_NE(HttpMethod("Trace"), trace);
+
+  EXPECT_NE(trace, HttpMethod(std::string()));
+  EXPECT_NE(HttpMethod(std::string()), trace);
+
+  EXPECT_NE(trace, HttpMethod::Get);
+  EXPECT_NE(trace, HttpMethod::Head);
+  EXPECT_NE(trace, HttpMethod::Post);
+  EXPECT_NE(trace, HttpMethod::Put);
+  EXPECT_NE(trace, HttpMethod::Delete);
+  EXPECT_NE(trace, HttpMethod::Patch);
+  EXPECT_NE(trace, HttpMethod("HUMUHUMUNUKUNUKUAPUAA"));
+
+  EXPECT_NE(HttpMethod::Get, trace);
+  EXPECT_NE(HttpMethod::Head, trace);
+  EXPECT_NE(HttpMethod::Post, trace);
+  EXPECT_NE(HttpMethod::Put, trace);
+  EXPECT_NE(HttpMethod::Delete, trace);
+  EXPECT_NE(HttpMethod::Patch, trace);
+  EXPECT_NE(HttpMethod("HUMUHUMUNUKUNUKUAPUAA"), trace);
+}
+
+TEST(HttpMethod, HawaiianFish)
+{
+  HttpMethod const humuhumunukunukuapuaa = HttpMethod("HUMUHUMUNUKUNUKUAPUAA");
```

Comment:
Lol :)

![image](https://user-images.githubusercontent.com/6527137/116600081-c158e780-a8dd-11eb-85c9-206d733b504c.png)

Reply:
Yes, a name that is unlikely to ever be picked for HTTP method :)

Reply:
My other candidate was `Eyjafjallajokull`, but in the end I thought it is slightly more likely to be picked for HTTP method name in the future (after 42th century).